circuits.js

```diff
@@ -97,6 +112,27 @@ function Circuits(options) {
     this.egressNodes = options.egressNodes;
 }
 
+Circuits.prototype.updateShorts = function updateShorts(shorts) {
+    this.shorts = shorts;
+    var keys = Object.keys(this.circuitsByServiceName);
+    for (var i = 0; i < keys.length; ++i) {
+        this.circuitsByServiceName[keys[i]].updateShorts(shorts);
+    }
+};
+
+Circuits.prototype.isShorted = function isShorted(callerName, serviceName, endpointName) {
+    if (!this.shorts) {
+        return false;
+    }
+    return this.shorts[callerName + '~' + serviceName + '~' + endpointName]
+        || this.shorts['*~' + serviceName + '~' + endpointName]
+        || this.shorts[callerName + '~*~' + endpointName]
+        || this.shorts[callerName + '~' + serviceName + '~*']
+        || this.shorts['*~*~' + endpointName]
+        || this.shorts['*~' + serviceName + '~*']
+        || this.shorts[callerName + '~*~*'];
```

Comment:
`Circuits#isShorted` is called on the hot request path at most once, at circuit creation time.  Thereafter it's called only at `remoteConfig` change time (the `updateShorts` path).